src/mpid/ch4/shm/ipc/gpu/gpu_pre.h

```diff
@@ -6,9 +6,27 @@
 #ifndef GPU_PRE_H_INCLUDED
 #define GPU_PRE_H_INCLUDED
 
+#include "uthash.h"
+
 typedef struct MPIDI_GPU_mem_handle {
     MPL_gpu_ipc_mem_handle_t ipc_handle;
-    int dev_id;
+    int global_dev_id;
 } MPIDI_GPU_mem_handle_t;
 
+typedef struct MPIDI_gpu_dev_id {
+    int local_dev_id;
+    int global_dev_id;
+    UT_hash_handle hh;
+} MPIDI_gpu_dev_id_t;
+
+typedef struct {
+    MPIDI_gpu_dev_id_t *local_to_global_map;
+    MPIDI_gpu_dev_id_t *global_to_local_map;
+    int **visible_dev_global_id;
+    int *local_ranks;
+    int *local_procs;
+} MPIDI_GPU_global_t;
+
+extern MPIDI_GPU_global_t MPIDI_gpu_global;
```

Comment:
These data structures are private and should not be exposed in the `gpu_pre.h` header (which is meant for the IPC layer).  They should go in some internal header file.